lib/nerve/statsd.rb

```diff
@@ -0,0 +1,31 @@
+require 'datadog/statsd'
+require 'nerve/log'
+
+module Nerve
+  module StatsD
+    def statsd
+      @@STATSD = StatsD.statsd_for(self.class.name) unless !@@STATSD_RELOAD && @@STATSD
+      @@STATSD
+    end
+
+    class << self
+      include Logging
+
+      @@STATSD_HOST = "localhost"
+      @@STATSD_PORT = 8125
+      @@STATSD_RELOAD = true
+
+      def statsd_for(classname)
+        log.debug "nerve: creating statsd client for class '#{classname}' on host '#{@@STATSD_HOST}' port #{@@STATSD_PORT}"
+        Datadog::Statsd.new(@@STATSD_HOST, @@STATSD_PORT)
+      end
+
+      def configure_statsd(opts)
+        @@STATSD_HOST = opts['host'] || @@STATSD_HOST
+        @@STATSD_PORT = (opts['port'] || @@STATSD_PORT).to_i
+        @@STATSD_RELOAD = true
```

Comment:
will this param ever be false?

Reply:
nice catch. I must change it after retrieving the client